public/pages/ConfigureModel/utils/helpers.ts

```diff
@@ -0,0 +1,346 @@
+      aggregationQuery: JSON.stringify(feature['aggregationQuery'], null, 4),
+      aggregationOf: get(
+        featureUiMetaData,
+        `${feature.featureName}.aggregationOf`
+      )
+        ? [
+            {
+              label: get(
+                featureUiMetaData,
+                `${feature.featureName}.aggregationOf`
+              ),
+            },
+          ]
+        : [],
+      featureType: get(featureUiMetaData, `${feature.featureName}.featureType`)
+        ? get(featureUiMetaData, `${feature.featureName}.featureType`)
+        : FEATURE_TYPE.CUSTOM,
+    };
+  });
+};
+
+export const focusOnFirstWrongFeature = (errors: any, setFieldTouched: any) => {
+  if (
+    //@ts-ignore
+    !!get(errors, 'featureList', []).filter((featureError) => featureError)
+      .length
+  ) {
+    const featureList = get(errors, 'featureList', []);
+    for (let i = featureList.length - 1; i >= 0; i--) {
+      if (featureList[i]) {
+        forOwn(featureList[i], function (value, key) {
+          setFieldTouched(`featureList.${i}.${key}`, true);
+        });
+        focusOnFeatureAccordion(i);
+      }
+    }
+    return true;
+  }
+  return false;
+};
+
+export const focusOnFeatureAccordion = (index: number) => {
+  const featureAccordion = document.getElementById(
+    `featureAccordionHeaders.${index}`
+  );
+  //@ts-ignore
+  featureAccordion.setAttribute('tabindex', '-1');
+  //@ts-ignore
+  featureAccordion.focus();
+  const header =
+    //@ts-ignore
+    featureAccordion.parentElement.parentElement.parentElement.parentElement;
+  //@ts-ignore
+  if (!header.className.includes('euiAccordion-isOpen')) {
+    //@ts-ignore
+    featureAccordion.click();
+  }
+};
+
+export const focusOnCategoryField = () => {
+  const component = document.getElementById('categoryFieldCheckbox');
+  component?.focus();
+};
+
+export const getCategoryFields = (dataTypes: DataTypes) => {
+  const keywordFields = get(dataTypes, 'keyword', []);
+  const ipFields = get(dataTypes, 'ip', []);
+  return keywordFields.concat(ipFields);
+};
+
+export const getShingleSizeFromObject = (
+  obj: object,
+  isHCDetector: boolean
+) => {
+  return get(
+    obj,
+    'shingleSize',
+    isHCDetector ? MULTI_ENTITY_SHINGLE_SIZE : SINGLE_ENTITY_SHINGLE_SIZE
+  );
+};
+
+export function clearModelConfiguration(ad: Detector): Detector {
+  return {
+    ...ad,
+    featureAttributes: [],
+    uiMetadata: {
+      ...ad.uiMetadata,
+      features: {},
+    },
+    categoryField: undefined,
+    shingleSize: SINGLE_ENTITY_SHINGLE_SIZE,
+  };
+}
+
+export function modelConfigurationToFormik(
+  detector: Detector
+): ModelConfigurationFormikValues {
+  const initialValues = cloneDeep(INITIAL_MODEL_CONFIGURATION_VALUES);
+  if (isEmpty(detector)) {
+    return initialValues;
+  }
+  return {
+    ...initialValues,
+    featureList: featuresToFormik(detector),
+    categoryFieldEnabled: !isEmpty(get(detector, 'categoryField', []))
+      ? true
+      : false,
```

Comment:
we can just use `!isEmpty(get(detector, 'categoryField', []))`?

Reply:
good catch, will remove